### ISSUE TITLE
Implementation details and difference at train/test time

### ISSUE THREAD

**Issue by ASalvail** [teganmaharaj/zoneout#3]
I was looking at `zoneout_seq2seq.py`, specifically at the `ZoneOut_Wrapper`. In addition to the fact that it doesn't compile (see Issue #2), there is some implementation details I would like details on.

From the `__call__` function, we have:

```python
if self.is_training:
  new_state = (1 - state_part_zoneout_prob) * tf.python.nn_ops.dropout(new_state_part - state_part, (1 - state_part_zoneout_prob), seed=self._seed) + state_part
else:
  new_state = state_part_zoneout_prob * state_part + (1 - state_part_zoneout_prob) * new_state_part
```

Assuming zoneout probability is p, we see that with probability (1-p) we zero out the a vector of the difference between the new state and the previous state, meaning that we end up with a vector of zeros and state update.

Question 1: why the multiplication by (1-p) ie `(1 - state_part_zoneout_prob)`. This basically reduce any update of the state by p%.
*EDIT: I answered this question myself, it's to counteract the implementation details of Dropout in Tensorflow.*
Question 2: I don't remember reading, in the paper, any use of using a weighted average at test time. How much does it help?
